python/lsst/sims/maf/metricBundles/metricBundle.py

```diff
@@ -119,9 +119,16 @@ def _setupMetricValues(self):
         # Can't store healpix slicer mask values in an int array.
         if dtype == 'int':
             dtype = 'float'
-        self.metricValues = ma.MaskedArray(data = np.empty(len(self.slicer), dtype),
-                                            mask = np.zeros(len(self.slicer), 'bool'),
-                                            fill_value= self.slicer.badval)
+        # Check if we have a 2D slicer
+        if 'bins' in self.slicer.slicePoints.keys():
```

Comment:
Oh - and actually, to sort of ease the way into this -- maybe we could just make slicers have shapes now? Instead of hacking this into "if there are 'bins' in the slicepoint?" 
self.slicer.shape = the shape for the data and mask that should come out of the slicer? and then for "normal" one-d slicers, we'd have self.slicer.shape = a simple number (the normal length of the slicer)